### PR TITLE
Make key fields mandatory for adding/updating views in CLI

### LINKED ISSUE #929: Make Key-fields necessary for Views
# Reason
It's difficult to explain what exactly composite views mean and in PgSql they are always required. So, let's just mandate that they are required for view

### PR DESCRIPTION
**Pull request by abhishekkumams**
## Why make this change?

- Closes #929 
  - To be consistent across all Dbs.

## What is this change?

- Added a validation check in CLI to mandate key-fields when adding views.

## How was this tested?
- [x] Unit Tests

### Adding a view without KeyFields
image

### Adding a view with KeyFields
image

### Updating a table to view without KeyFields
image

### Updating a table to view with KeyFields
image

# NOTE:
Engine will continue to support config with views that do not have key-fields provided in the config. This Validation is only applicable when adding/updating views through CLI.

### CODE REVIEW

File: src/Cli/Utils.cs
Comment (ayush3797) [inline src/Cli/Utils.cs:681]: nit: use `is` operator here.

**Review by ayush3797 — APPROVED** [review]

File: src/Cli.Tests/UpdateEntityTests.cs
Comment (seantleonard) [inline src/Cli.Tests/UpdateEntityTests.cs:1774]: I see you have a test case for when sourcetype isn't defined, can you also add a test to confirm failure where parameters and keyfields are included for a view?

File: src/Cli.Tests/AddEntityTests.cs
Comment (seantleonard) [inline src/Cli.Tests/AddEntityTests.cs:258]: can you add cases for when both keyfields and parameters are included for a view and table?

File: src/Cli/Utils.cs
Comment (seantleonard) [inline src/Cli/Utils.cs:681]: Even though this change is scoped to CLI to prevent breaking changes in the engine, it would be helpful to ensure (and add if missing) a test that validates the engine doesnt crash when supplied a view entity with missing key fields , especially for pgsql. So, if we ever decide to enforce this change in the engine, we will have to fix this test. The test would also prevent an accidental regression if we accidentally made this breaking change before being ready.

File: src/Cli/Utils.cs
Comment (seantleonard) [inline src/Cli/Utils.cs:683]: Because we are enforcing this for all database types, can you update the configuration/CLI documentation noting this behavior? This will add transparency and background context for non PG devs who may not expect keyfields to be required.

**Review by seantleonard — CHANGES_REQUESTED** [review]
looks good so far, pending tests/documentation to add before merge.

File: src/Cli/Utils.cs
Comment (Aniruddh25) [reply src/Cli/Utils.cs:681]: Actually, engine will still crash with `Primary key not configured on the given database object {tableName}` when key-fields are not specified for pg-sql. We should have a test for that. 

For certain views in MsSql, key-fields are optional since they can still be inferred - we need a test to ensure such views continue to be supported for MsSql - even though addition of such view through CLI would be resulting in validation failures.

Comment (abhishekkumams) [reply src/Cli/Utils.cs:681]: Added a Test for MsSQL, where in the config the views with no keyfields are supported.

File: src/Service.Tests/Configuration/ConfigurationTests.cs
Comment (seantleonard) [inline src/Service.Tests/Configuration/ConfigurationTests.cs:1101]: For graphQL, the response code doesn't necessarily tell you whether this worked. GraphQL requests can return 200 but can contain GraphQL errors. You'll need to check for a graphql error in the response. There should be GraphQL helper functions which can do this for you.

**Review by seantleonard — COMMENTED** [review]
need additional result checking for GraphQL request in your mssql view test. otherwise looks good. thank you for addressing comments.

File: src/Service.Tests/Configuration/ConfigurationTests.cs
Comment (Aniruddh25) [inline src/Service.Tests/Configuration/ConfigurationTests.cs:1041]:
```suggestion
        /// Engine supports config with some views that do not have keyfields specified in the config for MsSQL.
```

**Review by Aniruddh25 — APPROVED** [review]
LGTM, after the graphql response check

File: src/Service.Tests/Configuration/ConfigurationTests.cs
Comment (abhishekkumams) [reply src/Service.Tests/Configuration/ConfigurationTests.cs:1101]: updated at line 1104.

**Pull request merged**
